core/controllers/skill_editor.py

```diff
@@ -200,9 +201,28 @@ def delete(self, skill_id):
         skill_domain.Skill.require_valid_skill_id(skill_id)
         skill_ids_assigned_to_some_topic = (
             topic_services.get_all_skill_ids_assigned_to_some_topic())
```

Comment:
If you are going to delete it manually from topics anyway, then remove this check. Since 1, you can do that check inside your loop, and 2, this function also fetches all topics, so in this controller all topics are fetched twice which is inefficient.
Actually, I lean toward maybe moving the loop below to the domain layer into a `topic_services.delete_skill_from_all_topics()` function? Not sure about the last one.

Reply:
Yeah great optimization.
Done